tensorboard/plugins/projector/projector_plugin_test.py

```diff
@@ -193,13 +227,39 @@ def testPluginIsActive(self):
     self._GenerateProjectorTestData()
     self._SetupWSGIApp()
 
-    # Embedding data is available.
+    # The is_active method makes use of a separate thread, so we mock threading
+    # behavior to make this test deterministic.
+    tf.test.mock.patch('threading.Thread', FakeThread).start()
```

Comment:
This whole stubbing business is far more complicated than it needs to be. Why not just:
```py
patcher = tf.test.mock.patch('threading.Thread.start', autospec=True)
mock = patcher.start()
self.addCleanup(patcher.stop)  # remove `tearDown`

mock.assert_not_called()
self.assertFalse(self.plugin.is_active())
mock.assert_called_once_with(self.plugin._thread_for_determining_is_active)

self.assertFalse(self.plugin.is_active())
mock.assert_called_once_with(self.plugin._thread_for_determining_is_active)

self.plugin._thread_for_determining_is_active.run()
self.assertTrue(self.plugin.is_active())
```